master/buildbot/reporters/slack.py

```diff
@@ -0,0 +1,128 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+from future.utils import string_types
+
+from twisted.internet import defer
+
+from buildbot import config
+from datetime import datetime
+from humanize import naturaltime
+from buildbot.status.builder import SUCCESS
+from buildbot.util import httpclientservice
+from buildbot.reporters.http import HttpStatusPushBase
+
+SLACK_BASE_URL = "https://hooks.slack.com"
+
+
+class SlackStatusPush(HttpStatusPushBase):
```

Comment:
This should rather derive from notifierBase, which has more features for text based notifiers

Reply:
I had the worst development experience ever! Instead of simple NameErrors, I'd get a cryptic

```
	twisted.internet.defer.FirstError: FirstError[#0, [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.defer.FirstError'>: FirstError[#0, [Failure instance: Traceback: <class 'AttributeError'>: 'SlackPush' object has no attribute '_buildsetCompleteConsumer'
```

... and worst of all, `yield`s seem to swallow up even logging messages that are supposed to execute before them (when there's a silly error).

I sank maybe five hours, and endured much frustration in porting the code. I don't know if it's the difference between `HttpStatusPushBase` and `NotifierBase`, or if it's due to the recent Buildbot upgrade, but stuff needs to be fixed.